cpp/src/text/stemmer.cu

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/strings/detail/utilities.hpp>
+#include <cudf/strings/string_view.cuh>
+#include <cudf/strings/strings_column_view.hpp>
+#include <strings/utilities.cuh>
+
+#include <nvtext/stemmer.hpp>
+
+#include <thrust/for_each.h>
+#include <thrust/iterator/constant_iterator.h>
+
+namespace nvtext {
+namespace detail {
+namespace {
+
+/**
+ * @brief Return true if character at current iterator position
+ * is a consonant.
+ *
+ * A consonant is a letter other than a, e, i, o or u, and other
+ * than y preceded by a consonant.
+ *
+ * For `toy` the consonants are `t` and `y`, and in `syzygy` they
+ * are `s`, `z` and `g`.
+ *
+ * A _vowel_ is defined as _not a consonant_.
+ *
+ * @param string_iterator Iterator positioned to the character to check.
+ * @return True if the character at the iterator is a consonant.
+ */
+__device__ bool is_consonant(cudf::string_view::const_iterator string_iterator)
+{
+  auto ch = *string_iterator;
+  cudf::string_view const d_vowels("aeiou", 5);
+  if (d_vowels.find(ch) >= 0) return false;
+  if ((ch != 'y') || (string_iterator.position() == 0)) return true;
+  // for 'y' case, check previous character is a consonant
+  --string_iterator;
+  return d_vowels.find(*string_iterator) >= 0;
+}
+
+/**
+ * @brief Functor for the detail::is_letter_fn called to return true/false
+ * indicating the specified character is a consonant or a vowel.
+ */
+template <typename PositionIterator>
+struct is_letter_fn {
+  cudf::column_device_view const d_strings;
+  letter_type ltype;
+  PositionIterator position_itr;
+
+  __device__ bool operator()(cudf::size_type idx)
+  {
+    if (d_strings.is_null(idx)) return false;
+    auto const d_str = d_strings.element<cudf::string_view>(idx);
+    if (d_str.empty()) return false;
+    cudf::size_type position = position_itr[idx];
+    auto const length        = d_str.length();
```

Comment:
```suggestion
    auto const length  = d_str.length();
```

Reply:
I think this is because of clang

Reply:
This change is just to align the equal sign with that clang would do because of the change above (in case David decides to just commit the suggestion).